mnefun/_mnefun.py

```diff
@@ -1368,10 +1368,10 @@ def do_preprocessing_combined(p, subjects):
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
-            assert type(p.auto_bad_reject) and type(p.auto_bad_flat) == dict
+            flat = p.auto_bad_flat if isinstance(p.auto_bad_flat, dict) else None
```

Comment:
I'm actually not sure you need to make this change. Can't the user just set `p.auto_bad_flat = None`? It seems like that should work.